guides/asynchronous-tasks/readme.md

```diff
@@ -364,21 +366,23 @@ require 'async'
 require 'thread/local' # thread-local gem.
 
 class TimeCache
-	extend Thread::Local
+	extend Thread::Local # defines `instance` class method that lazy-creates a separate instance per thread
 	
 	def initialize
 		@current_time = nil
 	end
 	
-	def current_time
+	def current_time_string
 		refresh!
 		
 		return @current_time
 	end
 	
-	private def refresh!
+	private
+	
+	def refresh!
 		@refresh ||= Async(transient: true) do
-			while true
+			loop do
 				@current_time = Time.now.to_s
 				sleep(1)
 			end
```

Comment:
Re: the comment just below, I don't follow how an exception will get raised. I thought that would require us to call `.wait` on the task. Given that we're not doing that, doesn't `Async(transient: true) do ... end` return immediately without any chance of raising an exception when the reactor terminates all tasks?

And BTW, do you mean "terminates all transient tasks" there?

But also, I'm missing the significance of storing `nil` into the `@refresh` variable. The rest of the class doesn't use that. So the only effect would be if `refresh!` were called again, right? Was that the point?

Reply:
The `refresh!` task should exist while the reactor exists, if the user calls a top level `Async{TimeCache.current_time_string}` after exiting the top level block, the task created in `refresh!` will exit. We want to ensure that the next time `refresh!` is called, the task is recreated. Hence, on exiting the task, we ensure the instance variable is cleared.